tests/Feature/Http/Controllers/AnnounceControllerTest.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Tests\Feature\Http\Controllers;
+
+use App\Helpers\Bencode;
+use App\Models\Torrent;
+use App\Models\User;
+use GroupsTableSeeder;
+use Tests\TestCase;
+
+/**
+ * @see \App\Http\Controllers\AnnounceController
+ */
+class AnnounceControllerTest extends TestCase
+{
+    /**
+     * @test
+     */
+    public function announce_returns_an_ok_response()
+    {
+        $this->seed(GroupsTableSeeder::class);
+
+        $user = factory(User::class)->create([
+            'can_download' => true,
+        ]);
+
+        factory(Torrent::class)->create([
+            'info_hash' => bin2hex(md5('foo')),
```

Comment:
Should be a 20-byte SHA1 hash

Reply:
As with the previous comment, we don't check the requirement in the controller, so the actual value is irrelevant, but we'll add that to the test so we can assert against it in a future test.